src/EditorFeatures/Core.Wpf/InlineRename/UI/Adornment/RenameFlyout.xaml.cs

```diff
@@ -222,5 +231,25 @@ private void IdentifierTextBox_SelectionChanged(object sender, RoutedEventArgs e
             var startPoint = _viewModel.InitialTrackingSpan.GetStartPoint(buffer.CurrentSnapshot);
             _textView.SetSelection(new SnapshotSpan(startPoint + start, length));
         }
+
+        private void IdentifierTextBox_KeyDown(object sender, KeyEventArgs e)
+        {
+            // When smart rename is available, allow the user to scrolling up/down the suggestions by keyboard.
+            var smartRenameViewModel = _viewModel.SmartRenameViewModel;
+            if (smartRenameViewModel is not null)
+            {
+                var currentIdentifier = IdentifierTextBox.Text;
+                if (e.Key is Key.Down or Key.Up)
+                {
+                    var newIdentifier = smartRenameViewModel.ScrollSuggestions(currentIdentifier, down: e.Key == Key.Down);
+                    if (newIdentifier is not null)
+                    {
+                        _viewModel.IdentifierText = newIdentifier;
+                        IdentifierTextBox.Select(0, newIdentifier.Length);
```

Comment:
i'd prefer if we didn't select the whole thing.  that way i can add a suffix at the end of the suggested identifier.

Reply:
I changed the behavior here. Right now it would move the keyboard input cursor to the end the of the identifier